### ISSUE TITLE
cannot move location counter backwards (from 0000000008018958 to 0000000008008000)

### ISSUE THREAD

**Issue by SheperdHannigan** [RiS3-Lab/SHiFT#2]
Hi, there,

I'm trying to use the provided toolchain to reproduce the experiments. Like the tutorail projects, I cloned the projects using command "./cloner.sh -s Template-USBCDCH743FUZZER -d ./FuzzTestH743", and then I opened this  projects in my own STM32CubeIDE 1.16.0. And I configured the toolchain  named "gcc-arm-none-eabi-10.3-2021.12-0x24000000". And finally, I configured the GCC compiler with optional options "-fno-common", ""-fsanitize=address", "-fsanitize-coverage=trace-pc".  And I canceled the "fcyclomatic-complexity" that provided by STM32's toolchain.

After the above operations, I tried to build the tutorial projects, and got following information. Could you please help me with it?

```
gcc-arm-none-eabi-10.3-2021.12-0x24000000/bin/../lib/gcc/arm-none-eabi/10.3.1/../../../../arm-none-eabi/bin/ld:../STM32H745ZITX_FLASH_MPU_ASAN.ld:95 cannot move location counter backwards (from 0000000008018958 to 0000000008008000)
collect2: error: ld returned 1 exit status
```

**Comment by alejoseb** [comment]
Hi, I will check what is going on. Meanwhile, I noticed you are cloning the H743 template but the error message mentions the H745, those are different MCUs, are you sure you are following the instructions as detailed in the video?

**Comment by SheperdHannigan** [comment]
<img width="455" alt="1733337015990" src="https://github.com/user-attachments/assets/f10f9b00-5fb1-46c4-94ef-e4f1db086bc8">

Thank you so much for response.  The STM32H745ZITX_FLASH_MPU_ASAN is provided by Template-USBCDCH743FUZZER. I think that's why you said the error eessage mentions the H745.

**Comment by alejoseb** [comment]
I reproduced the steps described in the readme for the latest CUBEIDE 1.17.0 and it works as expected.
It does require to disable the fcyclomatic-complexity option as you also mentioned but nothing else.

I noticed that you " configured the GCC compiler with optional options "-fno-common", ""-fsanitize=address", "-fsanitize-coverage=trace-pc"  However,  notice that that configuration is done only for the test.c file and not for the entire project.
Also, after cloning the template with the provided instructions you do not need to configure anything else in your IDE besides the toolchain. The template is already configured with the "-fno-common", ""-fsanitize=address", "-fsanitize-coverage=trace-pc" for the test.c file.


![Screenshot from 2024-12-05 00-43-16](https://github.com/user-attachments/assets/80832d94-823d-4c84-b1b7-ab4d2cf026b1)
![Screenshot from 2024-12-05 00-45-19](https://github.com/user-attachments/assets/b8f1eaac-41b4-4bc6-af0d-a5b0562a43ad)
